site/app/views/course/CourseMaterialsView.php

```diff
@@ -176,12 +56,12 @@ public function listCourseMaterials($user) {
                 $priorities[$material] = 0.0;
             }
             $path = FileUtils::joinPaths($expected_path, $material);
-            $priorities[$material] += isset($json[$path]['sort_priority']) ? $json[$path]['sort_priority'] : $sort_default;
+            $priorities[$material] += isset($course_materials[$path]) ? $course_materials[$path]->getPriority() : $sort_default;
```

Comment:
The direct editing of the directory structure of course materials was a nightmare -- if you didn't do it perfectly you bricked the course materials page for your course.  I no longer have hope of supporting this.
All course materials edits should be done via the website (or eventually an API?).  All details should be in the database.  

We are transitioning away from the .json file -- we won't delete it in this PR though (just in case we have bugs).  All information about all course materials items will be in the database.  Any remnants of the original course materials design (that displayed stuff in the file system, even if it wasn't in the .json file) can be deleted.

Reply:
Alright. I think I will remove directory traversal 100% and have the database tell PHP what to read from filesystem. I saw in the code how you could upload a file and course materials would auto configure the json for it but I think switching to doing it through the website is the better option.